__tests__/Unit/Components/TaskList.test.tsx

```diff
@@ -0,0 +1,72 @@
+import TaskList from "@/components/tasks/TaskList/TaskList";
+import { render, screen, fireEvent } from "@testing-library/react";
+
+const TASK = {
+  id: "firestoreDocumentId123",
+  lossRate: {
+    dinero: 10,
+    neelam: 5,
+  },
+  links: ["https://realdevsquad.com/learn-site"],
+  completionAward: {
+    dinero: 110,
+    neelam: 10,
+  },
+  dependsOn: [],
+  assignee: "shmbajaj",
+  startedOn: "1618790400",
+  isNoteworthy: true,
+  title: "Testing and Determinsitic State",
+  purpose: "string",
+  percentCompleted: 0,
+  endsOn: "1618790400",
+  status: "progress",
+  featureUrl: "progress",
+  type: "feature",
+  createdBy: "shmbajaj",
+};
+const tasks = Array.from({ length: 10 }).map((_, index) => ({ ...TASK, id: TASK.id + index }));
+
+describe("TaskList", function () {
+  it("Should render TaskList", function () {
+    render(<TaskList tasks={tasks} />);
+
+    const _tasks = screen.queryAllByText(TASK.title);
+
+    expect(_tasks).toHaveLength(10);
+    expect(_tasks[0]).toBeInTheDocument();
+    expect(_tasks[0]).toHaveTextContent(TASK.title)
+  });
+
+  it("Should render see more button", function () {
+    render(<TaskList tasks={tasks} hasLimit={true} />);
+
+    const seeMore = screen.getByRole("button", { name: /see more/i });
+
+    expect(seeMore).toBeInTheDocument();
+  });
+
+  it("Should render 3 tasks intially and then render more 5 tasks after click event", function () {
+    render(<TaskList tasks={tasks} hasLimit={true} />);
+
+    const _tasks = screen.queryAllByText(TASK.title);
```

Comment:
any particular reason for using underscore before the variable name

Reply:
To differentiate between two set of tasks visually.

Reply:
Fine usually an underscore before a variable name is used to denote that the variable is private or only meant to be used internally. If possible we should modify it so that it doesn't cause confusion.

Reply:
I will change it in phase2.